mobile/src/modules/home/components/HomeStackedAreaChart.tsx

```diff
@@ -0,0 +1,159 @@
+
+import { Account, Transaction } from '@burstjs/core';
+import { convertBurstTimeToDate, convertNQTStringToNumber } from '@burstjs/util';
+import * as shape from 'd3-shape';
+import React from 'react';
+import { Text, TouchableOpacity, View } from 'react-native';
+import { Grid, StackedAreaChart, YAxis } from 'react-native-svg-charts';
+import { NavigationInjectedProps } from 'react-navigation';
+import { Button } from '../../../core/components/base/Button';
+import { Colors } from '../../../core/theme/colors';
+import { getBalanceHistoryFromTransactions } from '../../../core/utils/balance/getBalanceHistoryFromTransactions';
+import { BalanceHistoryItem } from '../../../core/utils/balance/typings';
+import { updateAccountTransactions } from '../../auth/store/actions';
+import { Metric, PriceInfo, PriceInfoReduxState } from '../../price-api/store/reducer';
+
+interface Props {
+  accounts: Account[];
+  priceApi: PriceInfoReduxState;
+}
+type TProps = Props;
+
+interface State {
+  priceInBTC: boolean
+}
+
+export class HomeStackedAreaChart extends React.PureComponent<TProps, State> {
+
+  state = {
+    selectedDateRange: 100,
+    priceInBTC: false
+  };
+
+  componentWillMount () {
+    // this.props.dispatch(updateAccountTransactionForMultipleAccounts(this.props.accounts));
+  }
+
+  calculateChartData () {
+    console.log(this.props.priceApi.historicalPriceInfo);
+    const accountTransactions = this.props.accounts.map((account) => {
+      if (account.transactions) {
+        return getBalanceHistoryFromTransactions(account.account,
+          convertNQTStringToNumber(account.balanceNQT), account.transactions);
+      }
+    });
+    const now = new Date();
+    const data = [];
+    const historicalPrice: Metric[] = [];
+    let i = 0;
+    for (const d = new Date(new Date().setDate(now.getDate() - this.state.selectedDateRange));
+      d <= now; d.setDate(d.getDate() + 1)) {
+      data[i] = {
+        day: new Date(d)
+      };
+
+      if (this.state.priceInBTC) {
+        const price = this.props.priceApi.historicalPriceInfo.Data.find((metric) => {
+          return this.sameDay(new Date(metric.time * 1000), d);
+        });
+        historicalPrice[d.getTime()] = price || this.props.priceApi.historicalPriceInfo.Data[0];
+      }
+      accountTransactions.map((transactions, accountIndex) => {
```

Comment:
actually, this should be transaction index, cuz it;s transactions array, isn't it?